libbindgen/src/ir/comp.rs

```diff
@@ -506,6 +509,7 @@ impl CompInfo {
         debug!("CompInfo::from_ty({:?}, {:?})", kind, cursor);
 
         let mut ci = CompInfo::new(kind);
+        ci.is_declaration = cursor.definition().unwrap().is_declaration();
```

Comment:
This is maybe not enough? The definition is _also_ a declaration, we need to search for a declaration that has no definition.

Note that clang has `CXCursor_NoDeclFound`, so we can probably do something like:

```rust
ci.is_forward_declaration = cursor.definition().map_or(true, |d| d.kind() == CXCursor_NoDeclFound);
```

That _might_ be enough.